### PR TITLE
Migrate test_contour to use dvc

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

As discussed in #1131, this PR moves the reference figures in fig.contour to use dvc.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by weiji14 — APPROVED** [review]

File: pygmt/tests/test_contour.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def test_contour_vec(region):
     y = y.flatten()
```
Comment (seisman) [inline pygmt/tests/test_contour.py:83]: @michaelgrund Although it makes no differences, could you also please update `projection="4i"` in the `test_contour_fail_no_data` test?

File: pygmt/tests/test_contour.py [hunk 2]
```diff
@@ -83,7 +83,7 @@ def test_contour_vec(region):
...
     z = (x - 0.5 * (region[0] + region[1])) ** 2 + 4 * y ** 2
     z = np.exp(-z / 10 ** 2 * np.log(2))
-    fig.contour(x=x, y=y, z=z, projection="X4i", region=region, frame="a", pen="")
+    fig.contour(x=x, y=y, z=z, projection="X10c", region=region, frame="a", pen="")
```
Comment (seisman) [inline pygmt/tests/test_contour.py:86]:
```suggestion
    fig.contour(x=x, y=y, z=z, projection="X10c", region=region, frame="a", pen=True)
```

File: pygmt/tests/test_contour.py [hunk 3]
```diff
@@ -93,7 +93,7 @@ def test_contour_matrix(data, region):
     Plot data.
     """
     fig = Figure()
-    fig.contour(data=data, projection="X3i", region=region, frame="ag", pen="")
+    fig.contour(data=data, projection="X10c", region=region, frame="ag", pen="")
```
Comment (seisman) [inline pygmt/tests/test_contour.py:96]:
```suggestion
    fig.contour(data=data, projection="X10c", region=region, frame="ag", pen=True)
```

File: pygmt/tests/test_contour.py (hunk 1 above)
Comment (seisman) [reply pygmt/tests/test_contour.py:83]: Also update `pen=""` to `pen=True`, as the latter is more Pythonic and readable.

Comment (michaelgrund) [reply pygmt/tests/test_contour.py:83]: Done, thanks for mentioning 😉 @seisman.

**Review by seisman — APPROVED** [review]
Looks great!

**Pull request merged**